pkgs/pkgs-lib/formats.nix

```
@@ -473,4 +473,56 @@ rec {
     '') {};
   };
 
+  envvars = {
+    export ? false
+  , listToStringSep ? ","
+  , listToStringFn ? lib.concatStringsSep listToStringSep
```

`listToStringSep` sounds like it takes a function, and actually it could be a function taking the attribute name as an argument.
I only understood it after reading the next two arguments.

I'd say something like this would be sufficient, but also consider adding the attribute name as an argument to both. An `Fn` suffix is not needed, and most higher order functions don't have it, so it's a bit unexpected.
```suggestion
  , listSep ? ","
  , listToString ? lib.concatStringsSep listSep
```

---

Might also be good to allow not supporting list values at all. I expect some programs to not support notions of multiple values.

`pkgs.formats.ini` works like this.

---

I think the `listToStringFn` might be overkill since environment variables can't have multiple values by themselves, but there are enviroment variables that have multiple values separated by a char (e.g. `$PATH` separated by `:`). It makes sense to me to use the module system to merge these values using a list, where the module creator can choose which character the program tends to use to concatenate such values.